share/functions/__fish_print_VBox_vms.fish

```diff
@@ -0,0 +1,28 @@
+function __fish_print_VBox_vms
+    set -l print_names true
+    set -l print_uuids true
+
+    if set -q argv[1]
+        if test $argv[1] = 'nouuids'
+            set print_uuids false
+        end
+        if test $argv[1] = 'nonames'
+            set print_names false
+        end
+    end
+
+    set -l lines (VBoxManage list vms | string match -r '"(.*)" {(.*)}')
+    while set -q lines[2]
+        if test $print_names = true
+            printf '%s\tVirtual machine\n' $lines[2]
+        end
+
+        if test $print_uuids = true
+            printf '%s\t%s virtual machine\n' $lines[3] $lines[2]
```

Comment:
>The only case were you get no output is when there are no virtual machines. Can you print an error without it being treated as a completion or messing with the buffer?

Anything sent to stderr won't be treated as completion, but it will mess with the buffer. The preferred mode of failure for completions is to just not print anything.

Anyway, this already handles no output (it never enters the while-loop), so as long as there's no way to get _different_ output it's okay.

>For the capitalized version, VBoxManage is present on Ubuntu, Arch Linux and macOS for the ones I checked. 

Okay then.